kong/db/schema/init.lua

```diff
@@ -1599,6 +1610,33 @@ function Schema:validate(input, full_check)
 end
 
 
+-- Iterate through input fields on update and check agianst schema for 
+-- immutable attribute. If immutable attribute is set, compare input values
+-- against entity values to detirmine whether input is valid.
+-- @param input The input table.
+-- @param entity The entity update will be performed on.
+-- @return True on success.
+-- On failure, it returns nil and a table containing all errors by field name.
+-- In all cases, the input table is untouched.
+function Schema:validate_immutable_fields(input, entity)
+  local errors = {}
+
+  input = tablex.deepcopy(input)
+
+  for key, field in pairs(input) do
+    if self.fields[key].immutable and entity[key] ~= nil and input[key] ~= entity[key] then
```

Comment:
Understand this PR is a draft implementation, but we'll need to assess how aggregate types integrate with `immutable` (not at all/they do but then we need more logic before asserting a table is identical to one already in DB, as comparing memory address is not enough in this context).

Reply:
The concept of this type of comparison was not established in the previous version of Kong but definitely seems necessary/needed in this re-implementation.  I see no reason why immutability should not apply to aggregate types as well.  Will look into a helper function which will assess likeness.